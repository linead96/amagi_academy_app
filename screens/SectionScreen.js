import React from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';
import Circle from '../components/SectionCircle';


const styles = StyleSheet.create({
  View: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    alignContent: 'center',
    paddingTop: 15,
    backgroundColor: '#fff',
  },
});

export default class SectionScreen extends React.Component {
  static navigationOptions = {
    title: 'Sections',
  };

  render() {
    const { navigation } = this.props;
    const { navigate } = navigation;

    return (
      <View style={styles.View}>
        <Circle color="#90F145" onPressButton={() => navigate('FlashCard')} />
        <Circle color="#2171D1" onPressButton={() => navigate('FlashCard')} />
        <Circle color="#CC003A" onPressButton={() => navigate('FlashCard')} />
        <Circle color="#67EE00" onPressButton={() => navigate('FlashCard')} />
        <Circle color="#4988D4" onPressButton={() => navigate('FlashCard')} />
        <Circle color="#F24576" onPressButton={() => navigate('FlashCard')} />
      </View>
    );
  }
}

SectionScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
