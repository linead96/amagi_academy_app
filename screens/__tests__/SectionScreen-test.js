import 'react-native';
import React from 'react';
import SectionScreen from '../SectionScreen';
import renderer from 'react-test-renderer';

describe('test SectionScreen functions and interface', () => {
    test('renders flash card initial screen', () => {
        const tree =renderer.create(<SectionScreen 
            navigation={{
                navigate: jest.fn(),
            }
        }/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
