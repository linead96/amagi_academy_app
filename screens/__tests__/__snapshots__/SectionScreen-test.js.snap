// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SectionScreen functions and interface renders flash card initial screen 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "paddingTop": 15,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#90F145",
        "borderRadius": 45,
        "height": 90,
        "margin": 30,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 90,
      }
    }
  />
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2171D1",
        "borderRadius": 45,
        "height": 90,
        "margin": 30,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 90,
      }
    }
  />
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#CC003A",
        "borderRadius": 45,
        "height": 90,
        "margin": 30,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 90,
      }
    }
  />
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#67EE00",
        "borderRadius": 45,
        "height": 90,
        "margin": 30,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 90,
      }
    }
  />
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#4988D4",
        "borderRadius": 45,
        "height": 90,
        "margin": 30,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 90,
      }
    }
  />
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F24576",
        "borderRadius": 45,
        "height": 90,
        "margin": 30,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 90,
      }
    }
  />
</View>
`;
